iot_config/src/gateway/service/mod.rs

```diff
@@ -286,21 +285,79 @@ impl iot_config::Gateway for GatewayService {
 
         tracing::debug!("fetching all gateways' info");
 
-        let pool = self.metadata_pool.clone();
+        let pool = self.pool.clone();
         let signing_key = self.signing_key.clone();
         let batch_size = request.batch_size;
         let region_map = self.region_map.clone();
 
         let (tx, rx) = tokio::sync::mpsc::channel(20);
 
         tokio::spawn(async move {
+            let epoch: DateTime<Utc> = "1970-01-01T00:00:00Z".parse().unwrap();
```

Comment:
chrono provides a constant for the epoch.
```
DateTime::UNIX_EPOCH;
```

Also, `DateTime::<Utc>::default()` is `UNIX_EPOCH`. We could consider passing `epoch` as an `Option<DateTime<Utc>>` to `stream_all_gateways_info` and `.unwrap_or_default()` when it's empty if that's more readable.

----

I don't know we're gaining anything from this function being called in a `tokio::select!` with a single branch.